### PR TITLE
[TECH-415][Service App Layer] Proto to Response

### PR DESCRIPTION
**Pull request by keerthikag**

### CODE REVIEW

File: hotel-search-api/src/main/java/com/hotel/api/search/controller/HotelSearchController.java [hunk 1]
```diff
@@ -21,7 +21,7 @@ public HotelSearchController(HotelAvailableService hotelAvailableService) {
     }
 
     @PostMapping(value = "/availability")
-    public HotelAvailabilityResponse getAvailableResponse(@RequestBody HotelAvailableRequest request) {
+    public String getAvailableResponse(@RequestBody HotelAvailableRequest request) {
         return hotelAvailableService.service(request);
```
Comment (LubaibaZ) [inline hotel-search-api/src/main/java/com/hotel/api/search/controller/HotelSearchController.java:25]: Please remove the produces Mediatype from @RequestMapping

File: hotel-search-api/src/main/java/com/hotel/api/search/adapter/HotelAvailabilityResponseAdapter.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import com.google.protobuf.util.JsonFormat;
+import com.hotel.service.availability.HotelAvailabilityResponse;
+
+import java.io.IOException;
+
+public class HotelAvailabilityResponseAdapter extends TypeAdapter<HotelAvailabilityResponse> {
+
```
Comment (LubaibaZ) [inline hotel-search-api/src/main/java/com/hotel/api/search/adapter/HotelAvailabilityResponseAdapter.java:14]: If possible add this as @component

**Review by LubaibaZ — CHANGES_REQUESTED** [review]

File: hotel-search-api/src/main/java/com/hotel/api/search/controller/HotelSearchController.java (hunk 1 above)
Comment (keerthikag) [reply hotel-search-api/src/main/java/com/hotel/api/search/controller/HotelSearchController.java:25]: Removed the produces mediatype

File: hotel-search-api/src/main/java/com/hotel/api/search/adapter/HotelAvailabilityResponseAdapter.java (hunk 2 above)
Comment (keerthikag) [reply hotel-search-api/src/main/java/com/hotel/api/search/adapter/HotelAvailabilityResponseAdapter.java:14]: added the annotation

**Review by LubaibaZ — APPROVED** [review]

**Pull request merged**